frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,428 @@
+                comments: "My comments",
+            };
+
+            let resp = await apiPOST(
+                `/public/postings/${posting.url_token}/submit`,
+                application,
+                true
+            );
+            expect(resp).toHaveStatus("success");
+
+            const applicationPrefillData = {
+                program: resp.payload.program,
+                department: resp.payload.department,
+                yip: resp.payload.yip,
+                previous_department_ta: resp.payload.previous_department_ta,
+                previous_university_ta: resp.payload.previous_university_ta,
+                previous_experience_summary:
+                    resp.payload.previous_experience_summary,
+                gpa: resp.payload.gpa,
+                comments: resp.payload.comments,
+            };
+
+            await switchToApplicantUser();
+
+            // Read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            expect(resp.payload["prefilled_data"]).toEqual(
+                expect.objectContaining(applicationPrefillData)
+            );
+            await restoreDefaultUser();
+        });
+
+        it(
+            "Submitting an application for a posting that does not have any positions or position preferences" +
+                " responds in error",
+            async () => {
+                // Create a new posting
+                const posting = {
+                    name: "CSC343F TA",
+                    intro_text: "Testing posting for CSC343F",
+                    open_date: "2021/04/01",
+                    close_date: "2021/05/01",
+                    availability: "open",
+                };
+                let resp = await apiPOST("/admin/postings", {
+                    ...posting,
+                    session_id: session.id,
+                });
+                expect(resp).toHaveStatus("success");
+                checkPropTypes(postingPropTypes, resp.payload);
+                expect(resp.payload.id).not.toBeNull();
+                Object.assign(posting, resp.payload);
+
+                await switchToApplicantUser();
+
+                // Submit an application without linking a position to the posting
+                const correctApplication = {
+                    answers: {
+                        ...applicant,
+                        position_preferences: {
+                            [position.position_code]: HIGH_PREFERENCE,
+                        },
+                    },
+                };
+
+                resp = await apiPOST(
+                    `/public/postings/${posting.url_token}/submit`,
+                    correctApplication,
+                    true
+                );
+                expect(resp).toHaveStatus("success");
```

Comment:
Shouldn't this be `"error"`?

Reply:
I thought it was supposed to be `"success"`. If that is not the case, there is an issue since the request has a status of `"success"` since this test does not fail.

Reply:
Well, the title of the test says it should error...you should make an issue for this.

Reply:
I have created an issue and commented this part of the test. The issue number is #630.